app/main.py

```diff
@@ -1,23 +1,33 @@
 import init_django_orm  # noqa: F401
 
-from django.db.models import QuerySet
+from django.db.models import QuerySet, Q, F
+from db.models import Customer, LoyaltyProgram, LoyaltyProgramParticipant
 
 
 def all_loyalty_program_names() -> QuerySet:
-    pass
+    return LoyaltyProgram.objects.values_list("name", "bonus_percentage")
 
 
 def not_active_customers() -> QuerySet:
-    pass
+    return LoyaltyProgramParticipant.objects.filter(
+        last_activity__year=2021
+    ).values("customer__first_name")
 
 
 def most_active_customers() -> QuerySet:
-    pass
+    return LoyaltyProgramParticipant.objects.order_by(
```

Comment:
This function actually seems less readable to me now.

Reply:
Accidentally mixed up two methods. I hope it's better now

Reply:
It looks better to me, but who am I to say that it's objectively better?